<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | 南开大学本科生毕业论文（设计）</title>
  <meta name="description" content="References | 南开大学本科生毕业论文（设计）" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="References | 南开大学本科生毕业论文（设计）" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="dongzhuoer/thesis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | 南开大学本科生毕业论文（设计）" />
  
  
  

<meta name="author" content="Zhuoer Dong" />


<meta name="date" content="2018-06-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="acknowledgement.html"/>

<script src="libs/header-attrs-2.5/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0/anchor-sections.js"></script>



<link rel="stylesheet" href="bookdown.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>封面</a></li>
<li class="chapter" data-level="" data-path="abstract-cn.html"><a href="abstract-cn.html"><i class="fa fa-check"></i>摘 要</a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html"><i class="fa fa-check"></i>一、前言和文献综述</a>
<ul>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#GEO"><i class="fa fa-check"></i>（一）GEO数据库</a></li>
<li class="chapter" data-level="" data-path="preface.html"><a href="preface.html#GSEA"><i class="fa fa-check"></i>（二）GSEA及其发展</a></li>
<li><a href="preface.html#LEM4">（三）<em>LEM4</em> 的功能</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="material-method.html"><a href="material-method.html"><i class="fa fa-check"></i>二、材料与方法</a>
<ul>
<li class="chapter" data-level="" data-path="material-method.html"><a href="material-method.html#data-software"><i class="fa fa-check"></i>（一）实验数据与软件</a></li>
<li class="chapter" data-level="" data-path="material-method.html"><a href="material-method.html#method"><i class="fa fa-check"></i>（二）实验方法</a>
<ul>
<li class="chapter" data-level="" data-path="material-method.html"><a href="material-method.html#rGEO"><i class="fa fa-check"></i>1. rGEO算法实现</a></li>
<li class="chapter" data-level="" data-path="material-method.html"><a href="material-method.html#qGSEA"><i class="fa fa-check"></i>2. qGSEA软件实现</a></li>
<li class="chapter" data-level="" data-path="material-method.html"><a href="material-method.html#GSEA-and-integration"><i class="fa fa-check"></i>3. GSEA及其结果的整合</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="material-method.html"><a href="material-method.html#diagram"><i class="fa fa-check"></i>（三）实验流程图</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html"><i class="fa fa-check"></i>三、实验数据与结果</a>
<ul>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html#probe2symbol"><i class="fa fa-check"></i>（一）rGEO——将探针对应到 HUGO 基因符号</a></li>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html#quick-run-GSEA"><i class="fa fa-check"></i>（二）qGSEA——快捷运行 GSEA</a></li>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html#GSEA-883-dataset"><i class="fa fa-check"></i>（三）在883项数据集中运行GSEA并整合结果</a>
<ul>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html#run-GSEA"><i class="fa fa-check"></i>1. 运行GSEA</a></li>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html#integrate-result"><i class="fa fa-check"></i>2. 整体分析GSEA的结果</a></li>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html#how-filter-geneset"><i class="fa fa-check"></i>3. 探索筛选基因集的方法</a></li>
<li class="chapter" data-level="" data-path="result.html"><a href="result.html#filter-and-explain"><i class="fa fa-check"></i>4. 筛选最终结果并分析其生物学意义</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="discuss.html"><a href="discuss.html"><i class="fa fa-check"></i>四、讨论与分析</a>
<ul>
<li class="chapter" data-level="" data-path="discuss.html"><a href="discuss.html#easy-use"><i class="fa fa-check"></i>（一）进一步增强易用性</a></li>
<li class="chapter" data-level="" data-path="discuss.html"><a href="discuss.html#more-platform"><i class="fa fa-check"></i>（二）支持更多的平台表格</a></li>
<li class="chapter" data-level="" data-path="discuss.html"><a href="discuss.html#merit-of-integration"><i class="fa fa-check"></i>（三）整合多项数据集的GSEA结果的优势</a></li>
<li><a href="discuss.html#better-filter">（四）更好地筛选与 <em>LEM4</em> 最相关的基因集</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i>附 录</a>
<ul>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html#appendix-plot"><i class="fa fa-check"></i>（一）附图</a></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html#appendix-table"><i class="fa fa-check"></i>（二）附表</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="acknowledgement.html"><a href="acknowledgement.html"><i class="fa fa-check"></i>致 谢</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">南开大学本科生毕业论文（设计）</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references hanging-indent">
<div>
<p>Agarwala, Richa, Tanya Barrett, Jeff Beck, Dennis A. Benson, Colleen Bollin, Evan Bolton, Devon Bourexis, et al. 2017. “Database Resources of the National Center for Biotechnology Information.” <em>Nucleic Acids Research</em> 45 (D1): D12–D17. <a href="https://doi.org/10.1093/nar/gkw1071">https://doi.org/10.1093/nar/gkw1071</a>.</p>
</div>
<div>
<p>Asencio, Claudio, Iain F. Davidson, Rachel Santarella-Mellwig, Thi Bach Nga Ly-Hartig, Moritz Mall, Matthew R. Wallenfang, Iain W. Mattaj, and Mátyás Gorjánácz. 2012. “Coordination of kinase and phosphatase activities by Lem4 enables nuclear envelope reassembly during mitosis.” <em>Cell</em> 150 (1): 122–35. <a href="https://doi.org/10.1016/j.cell.2012.04.043">https://doi.org/10.1016/j.cell.2012.04.043</a>.</p>
</div>
<div>
<p>Barrett, Tanya, Tugba O Suzek, Dennis B Troup, Stephen E Wilhite, Wing-Chi Ngau, Pierre Ledoux, Dmitry Rudnev, Alex E Lash, Wataru Fujibuchi, and Ron Edgar. 2005. “NCBI GEO: mining millions of expression profiles–database and tools.” <em>Nucleic Acids Research</em> 33 (Database issue): D562–6. <a href="https://doi.org/10.1093/nar/gki022">https://doi.org/10.1093/nar/gki022</a>.</p>
</div>
<div>
<p>Barrett, Tanya, Dennis B Troup, Stephen E Wilhite, Pierre Ledoux, Dmitry Rudnev, Carlos Evangelista, Irene F Kim, Alexandra Soboleva, Maxim Tomashevsky, and Ron Edgar. 2007. “NCBI GEO: mining tens of millions of expression profiles–database and tools update.” <em>Nucleic Acids Research</em> 35 (Database issue): D760–5. <a href="https://doi.org/10.1093/nar/gkl887">https://doi.org/10.1093/nar/gkl887</a>.</p>
</div>
<div>
<p>Barrett, Tanya, Dennis B. Troup, Stephen E. Wilhite, Pierre Ledoux, Dmitry Rudnev, Carlos Evangelista, Irene F. Kim, et al. 2009. “NCBI GEO: archive for high-throughput functional genomic data.” <em>Nucleic Acids Research</em> 37 (Database): D885–D890. <a href="https://doi.org/10.1093/nar/gkn764">https://doi.org/10.1093/nar/gkn764</a>.</p>
</div>
<div>
<p>Barrett, Tanya, Stephen E. Wilhite, Pierre Ledoux, Carlos Evangelista, Irene F. Kim, Maxim Tomashevsky, Kimberly A. Marshall, et al. 2012. “NCBI GEO: archive for functional genomics data sets—update.” <em>Nucleic Acids Research</em> 41 (D1): D991–D995. <a href="https://doi.org/10.1093/nar/gks1193">https://doi.org/10.1093/nar/gks1193</a>.</p>
</div>
<div>
<p>Barton, Lacy J., Alexey A. Soshnev, and Pamela K. Geyer. 2015. “Networking in the nucleus: A spotlight on LEM-domain proteins.” <em>Current Opinion in Cell Biology</em> 34: 1–8. <a href="https://doi.org/10.1016/j.ceb.2015.03.005">https://doi.org/10.1016/j.ceb.2015.03.005</a>.</p>
</div>
<div>
<p>Ben-Shaul, Yoram, Hagai Bergman, and Hermona Soreq. 2005. “Identifying subtle interrelated changes in functional gene categories using continuous measures of gene expression.” <em>Bioinformatics</em> 21 (7): 1129–37. <a href="https://doi.org/10.1093/bioinformatics/bti149">https://doi.org/10.1093/bioinformatics/bti149</a>.</p>
</div>
<div>
<p>Benson, Dennis A. 2004. “GenBank.” <em>Nucleic Acids Research</em> 33 (Database issue): D34–D38. <a href="https://doi.org/10.1093/nar/gki063">https://doi.org/10.1093/nar/gki063</a>.</p>
</div>
<div>
<p>Berk, Jason M., Kathryn E. Tifft, and Katherine L. Wilson. 2013. “The nuclear envelope LEM-domain protein emerin.” <em>Nucleus (Austin, Tex.)</em> 4 (4): 298–314. <a href="https://doi.org/10.4161/nucl.25751">https://doi.org/10.4161/nucl.25751</a>.</p>
</div>
<div>
<p>Bevilacqua, Vitoantonio, Paolo Pannarale, Mirko Abbrescia, Claudia Cava, Angelo Paradiso, and Stefania Tommasi. 2012. “Comparison of data-merging methods with SVM attribute selection and classification in breast cancer gene expression.” <em>BMC Bioinformatics</em> 13 Suppl 7 (Suppl 7): S9. <a href="https://doi.org/10.1186/1471-2105-13-S7-S9">https://doi.org/10.1186/1471-2105-13-S7-S9</a>.</p>
</div>
<div>
<p>Brachner, Andreas, and Roland Foisner. 2011. “Evolvement of LEM proteins as chromatin tethers at the nuclear periphery.” <em>Biochemical Society Transactions</em> 39 (6): 1735–41. <a href="https://doi.org/10.1042/BST20110724">https://doi.org/10.1042/BST20110724</a>.</p>
</div>
<div>
<p>Breslin, Thomas, Patrik Edén, and Morten Krogh. 2004. “Comparing functional annotation analyses with Catmap.” <em>BMC Bioinformatics</em> 5: 1–8. <a href="https://doi.org/10.1186/1471-2105-5-193">https://doi.org/10.1186/1471-2105-5-193</a>.</p>
</div>
<div>
<p>Cai, Mengli, Y Huang, R Ghirlando, K L Wilson, R Craigie, and G M Clore. 2001. “Solution structure of the constant region of nuclear envelope protein LAP2 reveals two LEM-domain structures: one binds BAF and the other binds DNA.” <em>The EMBO Journal</em> 20 (16): 4399–4407. <a href="https://doi.org/10.1093/emboj/20.16.4399">https://doi.org/10.1093/emboj/20.16.4399</a>.</p>
</div>
<div>
<p>Cai, Mengli, Ying Huang, Jeong Yong Suh, John M. Louis, Rodolfo Ghirlando, Robert Craigie, and G. Marius Clore. 2007. “Solution NMR structure of the barrier-to-autointegration factor-emerin complex.” <em>Journal of Biological Chemistry</em> 282 (19): 14525–35. <a href="https://doi.org/10.1074/jbc.M700576200">https://doi.org/10.1074/jbc.M700576200</a>.</p>
</div>
<div>
<p>Clark, Neil R., Maciej Szymkiewicz, Zichen Wang, Caroline D. Monteiro, Matthew R. Jones, and Avi Ma’ayan. 2015. “Principle Angle Enrichment Analysis (PAEA): Dimensionally reduced multivariate gene set enrichment analysis tool.” In <em>2015 Ieee International Conference on Bioinformatics and Biomedicine (Bibm)</em>, 155:256–62. 1. IEEE. <a href="https://doi.org/10.1109/BIBM.2015.7359689">https://doi.org/10.1109/BIBM.2015.7359689</a>.</p>
</div>
<div>
<p>Dinu, Irina, John D. Potter, Thomas Mueller, Qi Liu, Adeniyi J. Adewale, Gian S. Jhangri, Gunilla Einecke, Konrad S. Famulski, Philip Halloran, and Yutaka Yasui. 2007. “Improving gene set analysis of microarray data by SAM-GS.” <em>BMC Bioinformatics</em> 8: 1–13. <a href="https://doi.org/10.1186/1471-2105-8-242">https://doi.org/10.1186/1471-2105-8-242</a>.</p>
</div>
<div>
<p>Edgar, R. 2002. “Gene Expression Omnibus: NCBI gene expression and hybridization array data repository.” <em>Nucleic Acids Research</em> 30 (1): 207–10. <a href="https://doi.org/10.1093/nar/30.1.207">https://doi.org/10.1093/nar/30.1.207</a>.</p>
</div>
<div>
<p>Garber, Manuel, Manfred G. Grabherr, Mitchell Guttman, and Cole Trapnell. 2011. “Computational methods for transcriptome annotation and quantification using RNA-seq.” <em>Nature Methods</em> 8 (6): 469–77. <a href="https://doi.org/10.1038/nmeth.1613">https://doi.org/10.1038/nmeth.1613</a>.</p>
</div>
<div>
<p>Holden, Marit, Shiwei Deng, Leszek Wojnowski, and Bettina Kulle. 2008. “GSEA-SNP: applying gene set enrichment analysis to SNP data from genome-wide association studies.” <em>Bioinformatics (Oxford, England)</em> 24 (23): 2784–5. <a href="https://doi.org/10.1093/bioinformatics/btn516">https://doi.org/10.1093/bioinformatics/btn516</a>.</p>
</div>
<div>
<p>Irizarry, Rafael A, Chi Wang, Yun Zhou, and Terence P Speed. 2009. “Gene set enrichment analysis made simple.” <em>Statistical Methods in Medical Research</em> 18 (6): 565–75. <a href="https://doi.org/10.1177/0962280209351908">https://doi.org/10.1177/0962280209351908</a>.</p>
</div>
<div>
<p>Kim, Seon-Young, and David J Volsky. 2005. “PAGE: parametric analysis of gene set enrichment.” <em>BMC Bioinformatics</em> 6: 144. <a href="https://doi.org/10.1186/1471-2105-6-144">https://doi.org/10.1186/1471-2105-6-144</a>.</p>
</div>
<div>
<p>Li, Junan, Anjali Mahajan, and Ming-Daw Tsai. 2006. “Ankyrin Repeat: A Unique Motif Mediating Protein−Protein Interactions †.” <em>Biochemistry</em> 45 (51): 15168–78. <a href="https://doi.org/10.1021/bi062188q">https://doi.org/10.1021/bi062188q</a>.</p>
</div>
<div>
<p>Liberzon, Arthur, Chet Birger, Helga Thorvaldsdóttir, Mahmoud Ghandi, Jill P. Mesirov, and Pablo Tamayo. 2015. “The Molecular Signatures Database (MSigDB) hallmark gene set collection.” <em>Cell Systems</em> 1 (6): 417–25. <a href="https://doi.org/10.1016/j.cels.2015.12.004">https://doi.org/10.1016/j.cels.2015.12.004</a>.</p>
</div>
<div>
<p>Liberzon, Arthur, Aravind Subramanian, Reid Pinchback, Helga Thorvaldsdóttir, Pablo Tamayo, and Jill P. Mesirov. 2011. “Molecular signatures database (MSigDB) 3.0.” <em>Bioinformatics (Oxford, England)</em> 27 (12): 1739–40. <a href="https://doi.org/10.1093/bioinformatics/btr260">https://doi.org/10.1093/bioinformatics/btr260</a>.</p>
</div>
<div>
<p>Love, Michael I., Wolfgang Huber, and Simon Anders. 2014. “Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2.” <em>Genome Biology</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.</p>
</div>
<div>
<p>Maglott, D., Gota Morota, KD Pruitt, T Tatusova, P Kell, S Duvick, EP Spalding, et al. 2004. “Entrez Gene: gene-centered information at NCBI.” <em>Nucleic Acids Research</em> 33 (Database issue): D54–D58. <a href="https://doi.org/10.1093/nar/gki031">https://doi.org/10.1093/nar/gki031</a>.</p>
</div>
<div>
<p>Nakamura, Yasukazu, Guy Cochrane, and Ilene Karsch-Mizrachi. 2013. “The International Nucleotide Sequence Database Collaboration.” <em>Nucleic Acids Research</em> 41 (D1): D21–D24. <a href="https://doi.org/10.1093/nar/gks1084">https://doi.org/10.1093/nar/gks1084</a>.</p>
</div>
<div>
<p>Nam, Dougu, and Seon Young Kim. 2008. “Gene-set approach for expression pattern analysis.” <em>Briefings in Bioinformatics</em> 9 (3): 189–97. <a href="https://doi.org/10.1093/bib/bbn001">https://doi.org/10.1093/bib/bbn001</a>.</p>
</div>
<div>
<p>Nikolsky, Yuri, Evgeny Sviridov, Jun Yao, Damir Dosymbekov, Vadim Ustyansky, Valery Kaznacheev, Zoltan Dezso, et al. 2008. “Genome-wide functional synergy between amplified and mutated genes in human breast cancer.” <em>Cancer Research</em> 68 (22): 9532–40. <a href="https://doi.org/10.1158/0008-5472.CAN-08-3082">https://doi.org/10.1158/0008-5472.CAN-08-3082</a>.</p>
</div>
<div>
<p>O’Leary, Nuala A., Mathew W. Wright, J. Rodney Brister, Stacy Ciufo, Diana Haddad, Rich McVeigh, Bhanu Rajput, et al. 2016. “Reference sequence (RefSeq) database at NCBI: Current status, taxonomic expansion, and functional annotation.” <em>Nucleic Acids Research</em> 44 (D1): D733–D745. <a href="https://doi.org/10.1093/nar/gkv1189">https://doi.org/10.1093/nar/gkv1189</a>.</p>
</div>
<div>
<p>Pan, K.-H., C.-J. Lih, and S. N. Cohen. 2005. “Effects of threshold choice on biological conclusions reached during analysis of gene expression by DNA microarrays.” <em>Proceedings of the National Academy of Sciences</em> 102 (25): 8961–5. <a href="https://doi.org/10.1073/pnas.0502674102">https://doi.org/10.1073/pnas.0502674102</a>.</p>
</div>
<div>
<p>Peng, Shaoliang, Shunyun Yang, Xiaochen Bo, and Fei Li. 2017. “paraGSEA: a scalable approach for large-scale gene expression profiling.” <em>Nucleic Acids Research</em> 45 (17): e155–e155. <a href="https://doi.org/10.1093/nar/gkx679">https://doi.org/10.1093/nar/gkx679</a>.</p>
</div>
<div>
<p>Pontius, J U, L Wagner, and G D Schuler. 2003. “UniGene: a unified view of the transcriptome.” <em>The NCBI Handbook</em>, no. 21: 1–12. <a href="https://www.ncbi.nlm.nih.gov/books/NBK21083/">https://www.ncbi.nlm.nih.gov/books/NBK21083/</a>.</p>
</div>
<div>
<p>Segura-Totten, Miriam, and Katherine L. Wilson. 2004. “BAF: roles in chromatin, nuclear structure and retrovirus integration.” <em>Trends in Cell Biology</em> 14 (5): 261–6. <a href="https://doi.org/10.1016/j.tcb.2004.03.004">https://doi.org/10.1016/j.tcb.2004.03.004</a>.</p>
</div>
<div>
<p>Simillion, Cedric, Robin Liechti, Heidi E. L. Lischer, Vassilios Ioannidis, and Rémy Bruggmann. 2017. “Avoiding the pitfalls of gene set enrichment analysis with SetRank.” <em>BMC Bioinformatics</em> 18 (1): 151. <a href="https://doi.org/10.1186/s12859-017-1571-6">https://doi.org/10.1186/s12859-017-1571-6</a>.</p>
</div>
<div>
<p>Snyers, Luc, Renate Erhart, Sylvia Laffer, Oliver Pusch, Klara Weipoltshammer, and Christian Schöfer. 2018. “LEM4/ANKLE-2 deficiency impairs post-mitotic re-localization of BAF, LAP2<span class="math inline">\(\alpha\)</span> and LaminA to the nucleus, causes nuclear envelope instability in telophase and leads to hyperploidy in HeLa cells.” <em>European Journal of Cell Biology</em> 97 (1): 63–74. <a href="https://doi.org/10.1016/j.ejcb.2017.12.001">https://doi.org/10.1016/j.ejcb.2017.12.001</a>.</p>
</div>
<div>
<p>Subramanian, Aravind, Heidi Kuehn, Joshua Gould, Pablo Tamayo, and Jill P. Mesirov. 2007. “GSEA-P: a desktop application for Gene Set Enrichment Analysis.” <em>Bioinformatics</em> 23 (23): 3251–3. <a href="https://doi.org/10.1093/bioinformatics/btm369">https://doi.org/10.1093/bioinformatics/btm369</a>.</p>
</div>
<div>
<p>Subramanian, A., P. Tamayo, V. K. Mootha, S. Mukherjee, B. L. Ebert, M. A. Gillette, A. Paulovich, et al. 2005. “Gene set enrichment analysis: A knowledge-based approach for interpreting genome-wide expression profiles.” <em>Proceedings of the National Academy of Sciences</em> 102 (43): 15545–50. <a href="https://doi.org/10.1073/pnas.0506580102">https://doi.org/10.1073/pnas.0506580102</a>.</p>
</div>
<div>
<p>Wagner, Nicole, and Georg Krohne. 2007. “LEM-Domain Proteins: New Insights into Lamin-Interacting Proteins.” In <em>International Review of Cytology</em>, 261:1–46. <a href="https://doi.org/10.1016/S0074-7696(07)61001-8">https://doi.org/10.1016/S0074-7696(07)61001-8</a>.</p>
</div>
<div>
<p>Wang, Weichen, Zhiyi Qin, Zhixing Feng, Xi Wang, and Xuegong Zhang. 2013. “Identifying differentially spliced genes from two groups of RNA-seq samples.” <em>Gene</em> 518 (1): 164–70. <a href="https://doi.org/10.1016/j.gene.2012.11.045">https://doi.org/10.1016/j.gene.2012.11.045</a>.</p>
</div>
<div>
<p>Wang, Xi, and Murray J. Cairns. 2014. “SeqGSEA: a Bioconductor package for gene set enrichment analysis of RNA-Seq data integrating differential expression and splicing.” <em>Bioinformatics</em> 30 (12): 1777–9. <a href="https://doi.org/10.1093/bioinformatics/btu090">https://doi.org/10.1093/bioinformatics/btu090</a>.</p>
</div>
<div>
<p>Yates, Bethan, Bryony Braschi, Kristian A. Gray, Ruth L. Seal, Susan Tweedie, and Elspeth A. Bruford. 2017. “Genenames.org: the HGNC and VGNC resources in 2017.” <em>Nucleic Acids Research</em> 45 (D1): D619–D625. <a href="https://doi.org/10.1093/nar/gkw1033">https://doi.org/10.1093/nar/gkw1033</a>.</p>
</div>
<div>
<p>Yoon, Sora, Seon-Young Kim, and Dougu Nam. 2016. “Improving Gene-Set Enrichment Analysis of RNA-Seq Data with Small Replicates.” Edited by Dongmei Li. <em>PLOS ONE</em> 11 (11): e0165919. <a href="https://doi.org/10.1371/journal.pone.0165919">https://doi.org/10.1371/journal.pone.0165919</a>.</p>
</div>
<div>
<p>Zerbino, Daniel R., Premanand Achuthan, Wasiu Akanni, M Ridwan Amode, Daniel Barrell, Jyothish Bhai, Konstantinos Billis, et al. 2018. “Ensembl 2018.” <em>Nucleic Acids Research</em> 46 (D1): D754–D761. <a href="https://doi.org/10.1093/nar/gkx1098">https://doi.org/10.1093/nar/gkx1098</a>.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="acknowledgement.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "sepia",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/dongzhuoer/thesis/edit/bookdown/index.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "none"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
